package/MDAnalysis/analysis/nucleicacids.py

```diff
@@ -209,24 +253,147 @@ class WatsonCrickDist(NucPairDist):
        MDAnalysis 2.5.0. Please use the class attribute :attr:`times` instead.
     """
 
-    def __init__(self, strand1: List[Residue], strand2: List[Residue],
+    def __init__(self, strand1: List[ResidueClass], strand2: List[ResidueClass],
                  n1_name: str = 'N1', n3_name: str = "N3",
                  g_name: str = 'G', a_name: str = 'A', u_name: str = 'U',
                  t_name: str = 'T', c_name: str = 'C',
                  **kwargs) -> None:
-        sel1: List[mda.AtomGroup] = []
-        sel2: List[mda.AtomGroup] = []
-        strand = zip(strand1, strand2)
 
-        for s in strand:
-            if s[0].resname[0] in [c_name, t_name, u_name]:
-                a1, a2 = n3_name, n1_name
-            elif s[0].resname[0] in [a_name, g_name]:
-                a1, a2 = n1_name, n3_name
-            else:
-                raise ValueError(f"{s} are not valid nucleic acids")
+        selections: Tuple[List[mda.AtomGroup], List[mda.AtomGroup]] = self.select_strand_atoms(
+            strand1, strand2, n1_name, n3_name, 
+            g_name=g_name, a_name=a_name,
+            t_name=t_name, u_name=u_name, c_name=c_name
+        )
+
+        super(WatsonCrickDist, self).__init__(selections[0], selections[1], **kwargs)
+
+
+class MinorPairDist(NucPairDist):
+    r"""Minor-Pair basepair distance for selected residues over a trajectory.
+
+    Takes two lists of :class:`~MDAnalysis.core.groups.Residue` objects and
+    calculates the Watson-Crick distance between them over the trajectory.
+    Bases are matched by their index in the lists given as arguments.
+
+    Parameters
+    ----------
+    strand1: List[Residue]
+        First list of bases
+    strand2: List[Residue]
+        Second list of bases
+    o2_name: str (optional)
+        Name of Oxygen 2 of nucleic acids
+        by default assigned to O2
+    c2_name: str (optional)
+        Name of Carbon 2 of nucleic acids
+        by default assigned to C2
+    g_name: str (optional)
+        Name of Guanine in topology
+        by default assigned to G
+    a_name: str (optional)
+        Name of Adenine in topology
+        by default assigned to G
+    u_name: str (optional)
+        Name of Uracil in topology
+        by default assigned to U
+    t_name: str (optional)
+        Name of Thymine in topology
+        by default assigned to T
+    c_name: str (optional)
+        Name of Cytosine in topology
+        by default assigned to C
+    **kwargs: dict
+        arguments for :class:`~MDAnalysis.analysis.base.AnalysisBase`
+
+    Attributes
+    ----------
```

Comment:
add docs for `times` thanks to AnalysisBase

Reply:
done